src/Framework/Framework/Configuration/CustomPrimitiveTypeAttribute.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace DotVVM.Framework.Configuration;
+
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct)]
+public class CustomPrimitiveTypeAttribute : Attribute
+{
+    /// <summary>
+    /// Gets a type which will be used for this custom type on the client side. Only types from ReflectionUtils.PrimitiveTypes (or their nullable versions) are supported.
```

Comment:
This type can not be actually nullable. We handle those differently

Reply:
What is the problem with nullable types here? In the sample, we are using `Guid?` since the typed id is a reference type and can be null.

Reply:
This use-case seems to work nicely with reference types. If `SampleId`, which is a reference type, maps to `Guid?` - then properties that are of type `SampleId` work correctly.

The problem comes when `SampleId` is a value type (struct). Then we can have also a nullable and non-nullable property. This can generate funny stuff like this: 
![image](https://user-images.githubusercontent.com/12575176/204799987-dd789c5f-f1a9-4c05-aa02-63e386ac8451.png)

Reply:
We should probably add a check when generating metadata for value types so that such situation can not happen. Otherwise users should not registrate these with nullable underlying type, otherwise it can be doubled.